spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/support/JdbcRepositoryFactoryBean.java

```diff
@@ -47,11 +53,14 @@
 		extends TransactionalRepositoryFactoryBeanSupport<T, S, ID> implements ApplicationEventPublisherAware {
 
 	private ApplicationEventPublisher publisher;
+	private ConfigurableListableBeanFactory beanFactory;
 	private RelationalMappingContext mappingContext;
 	private RelationalConverter converter;
 	private DataAccessStrategy dataAccessStrategy;
 	private QueryMappingConfiguration queryMappingConfiguration = QueryMappingConfiguration.EMPTY;
 	private NamedParameterJdbcOperations operations;
+	private String dataAccessStrategyRef;
```

Comment:
Let's not use string bean names in here. Rather `afterPropertiesSet()` should try to find a default bean using the type. Calling `BeanFactory.getBean(Class<?>)` will take care of `@Primary` selection and will report the appropriate exception if the bean is not unique.